### PR TITLE
Bump node12 to node16

### PR DESCRIPTION
**Pull request by quinnjn**
Bumps node12 to node16. Also suggesting at least node 16 for install. Node 12 is EOL

### CODE REVIEW

**Comment by javierjulio** [comment]
Thanks @quinnjn! This will resolve the reported warning in #29. Sorry, I didn't realize there was a PR before creating the issue. 

The GHA workflows should be updated as well to both use `ubuntu-latest` and at least have a `node-version: '16.x'` set, unless v18 is preferred.

File: .github/workflows/pr-branch-checks.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       - uses: actions/setup-node@v1
         name: setup node
         with:
-          node-version: '13.x'
+          node-version-file: '.nvmrc'
```
Comment (cakeinpanic) [inline .github/workflows/pr-branch-checks.yml:17]: wow, I did not know this feature existed, cool!

**Review by cakeinpanic — COMMENTED** [review]
Great idea!
But would this action work then for people who are using node below 16 in their pipelines?

File: .github/workflows/pr-branch-checks.yml (hunk 1 above)
Comment (quinnjn) [reply .github/workflows/pr-branch-checks.yml:17]: So nice right?! It should be the default!

File: action.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ inputs:
       required: false
       default: 'false'
 runs:
-  using: 'node12'
+  using: 'node16'
```
[the remark this answers is not in the document]
Comment (quinnjn) [reply action.yml:34]: I don't think. The action should be an isolated step so it shouldn't matter what node version it is.

Regardless [node 12 is EOL](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V12.md#notable-changes) and should be upgraded.

**Review by cakeinpanic — APPROVED** [review]

**Pull request merged**